### PR TITLE
QC-1160 Update flag-related nomenclature

### PR DESCRIPTION
**Pull request by knopers8**
There are some forward declarations that I had to get rid of for the time being, since C++ does not like combining aliases with forward declarations. They can be brought back after the migration is complete.

### CODE REVIEW

File: Framework/include/QualityControl/ObjectMetadataKeys.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ constexpr auto qcTaskName = "qc_task_name";
 constexpr auto qcTaskClass = "qc_task_class";
 constexpr auto qcQuality = "qc_quality";
 constexpr auto qcCheckName = "qc_check_name";
-constexpr auto qcTRFCName = "qc_trfc_name";
+constexpr auto qcQCFCName = "qc_qcfc_name";
```
Comment (Barthelemy) [inline Framework/include/QualityControl/ObjectMetadataKeys.h:38]: I guess that it does not need to be updated in the QCDB ?

File: Framework/test/testQualitiesToFlagCollectionConverter.cxx [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    { Quality::Good, "xyzCheck", "DET", {}, {}, {}, { { metadata_keys::validFrom, "5" }, { metadata_keys::validUntil, "10" } } },
+    { Quality::Good, "xyzCheck", "DET", {}, {}, {}, { { metadata_keys::validFrom, "10" }, { metadata_keys::validUntil, "14" } } },
+    { Quality::Good, "xyzCheck", "DET", {}, {}, {}, { { metadata_keys::validFrom, "15" }, { metadata_keys::validUntil, "49" } } },
+    { Quality::Bad, "xyzCheck", "DET", {}, {}, {}, { { metadata_keys::validFrom, "50" }, { metadata_keys::validUntil, "80" } } },
+    { Quality::Bad, "xyzCheck", "DET", {}, {}, {}, { { metadata_keys::validFrom, "80" }, { metadata_keys::validUntil, "95" } } },
+    { Quality::Bad, "xyzCheck", "DET", {}, {}, {}, { { metadata_keys::validFrom, "96" }, { metadata_keys::validUntil, "120" } } }
+  };
+  std::unique_ptr<QualityControlFlagCollection> qcfc{ new QualityControlFlagCollection("test1", "DET", { 5, 100 }) };
+  QualitiesToFlagCollectionConverter converter(std::move(qcfc), "qc/DET/QO/xyzCheck");
+  ;
```
Comment (Barthelemy) [inline Framework/test/testQualitiesToFlagCollectionConverter.cxx:201]: typo ?

**Review by Barthelemy — APPROVED** [review]
I went through and see that it is only name substitution. I did not spot any mistakes.

File: Framework/include/QualityControl/ObjectMetadataKeys.h (hunk 1 above)
Comment (knopers8) [reply Framework/include/QualityControl/ObjectMetadataKeys.h:38]: no, anyway it's prototype code.

File: Framework/test/testQualitiesToFlagCollectionConverter.cxx (hunk 2 above)
Comment (knopers8) [reply Framework/test/testQualitiesToFlagCollectionConverter.cxx:201]: i'm sorry, where?

Comment (Barthelemy) [reply Framework/test/testQualitiesToFlagCollectionConverter.cxx:201]: the semicolon all alone on a line

Comment (knopers8) [reply Framework/test/testQualitiesToFlagCollectionConverter.cxx:201]: ah yes of course

**Pull request merged**